api/middleware.js

```diff
@@ -0,0 +1,54 @@
+const express = require('express');
+const morgan = require('morgan');
+const helmet = require('helmet');
+const session = require('express-session');
+const KnexSessionStore = require('connect-session-knex')(session);
+
+const restrictPath = (path = '/api/restricted') => (req, res, next) => {
```

Comment:
Nice job using a conditional statement with you protected function.  This is another way to write it ```function protected(req, res, next) {
    if (req.session && req.session.username) {
        next();
    } else {
        return res.status(401).json({error:'Incorrect Credentials'});
    }
}```